### PR TITLE
Parse Breadcrumbs for OnSend Callbacks

### PR DESCRIPTION
**Pull request by lemnik**
## Goal
Parse `BreadcrumbInternal` objects (which will be wrapped in `Breadcrumb` objects) from stored events, so that they can be processed by registered `OnSendCallback`s.

## Testing
Added a deserialization test to `BreadcrumbSerializationTest`

### CODE REVIEW

File: bugsnag-android-core/src/main/java/com/bugsnag/android/BreadcrumbType.kt [hunk 1]
```diff
@@ -38,4 +38,8 @@ enum class BreadcrumbType(private val type: String) {
     USER("user");
 
     override fun toString() = type
+
+    internal companion object {
+        internal fun fromDescriptor(type: String) = BreadcrumbType.values().find { it.type == type }
```
Comment (fractalwrench) [inline bugsnag-android-core/src/main/java/com/bugsnag/android/BreadcrumbType.kt:43]:
```suggestion
        internal fun fromDescriptor(type: String) = values().singleOrNull { it.type == type }
```

File: bugsnag-android-core/src/test/java/com/bugsnag/android/BreadcrumbSerializationTest.kt [hunk 2]
```diff
@@ -17,7 +17,7 @@ internal class BreadcrumbSerializationTest {
             val timestamp = Date(0)
             return generateSerializationTestCases(
                 "breadcrumb",
-                Breadcrumb("hello world", BreadcrumbType.MANUAL, mutableMapOf(), timestamp, NoopLogger),
+                Breadcrumb("helloworld", BreadcrumbType.MANUAL, mutableMapOf(), timestamp, NoopLogger),
```
Comment (fractalwrench) [inline bugsnag-android-core/src/test/java/com/bugsnag/android/BreadcrumbSerializationTest.kt:20]: I'm not sure why this has changed?

Comment (lemnik) [reply bugsnag-android-core/src/test/java/com/bugsnag/android/BreadcrumbSerializationTest.kt:20]: The JSON comparison as it currently stands on this branch removes *all* whitespace before comparing.
The issue has been fixed on `next` (https://github.com/bugsnag/bugsnag-android/pull/1512) and this code will change back before we merge the integration branch down.

But to keep the rebase / PRs simple I opted to simply change the same model to match the data in the JSON fixtures. Once the integration branch is ready to be merged into next - these tests will start failing and need to be fixed "back" again.

File: bugsnag-android-core/src/main/java/com/bugsnag/android/BugsnagEventMapper.kt [hunk 3]
```diff
@@ -11,6 +11,19 @@ internal class BugsnagEventMapper(
     private val logger: Logger
 ) {
 
+    @Suppress("UNCHECKED_CAST")
+    internal fun convertBreadcrumbInternal(breadcrumb: Map<String, Any?>): BreadcrumbInternal {
+        return BreadcrumbInternal(
+            breadcrumb.readEntry("name"),
+            breadcrumb.readEntry<String>("type").let { type ->
+                BreadcrumbType.fromDescriptor(type)
+                    ?: throw IllegalArgumentException("unknown BreadcrumbType: '$type'")
```
Comment (kattrali) [inline bugsnag-android-core/src/main/java/com/bugsnag/android/BugsnagEventMapper.kt:20]: I'd coerce unknown types to `manual`, that's how Potential Shenanigans are handled on other platforms / in request data.

**Review by kattrali — APPROVED** [review]

**Review by fractalwrench — APPROVED** [review]
LGTM

**Pull request merged**